src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterDownload.java

```diff
@@ -914,7 +917,7 @@ private Resource writeRaster(
         @SuppressWarnings("PMD.CloseResource") // This stream will be properly closed
         // when closing the os variable
         final ImageOutputStream fileImageOutputStreamExtImpl =
-                new ImageOutputStreamAdapter(output.out());
+                new FileImageOutputStreamExtImpl(output.file(), BUFFER_SIZE);
```

Comment:
In theory, a plug-in could replace the DefaultProcessArtifacsStore and the output might not be on the file system... but we don't have an actual implemention of it for the time being (e..g, could be a S3 based one for example), and even in that case it would be probably better to write on the local file system and then copy over (which would be the responsibility of the artifacts store implementation I suppose).